cmd/kubeadm/app/preflight/checks.go

```diff
@@ -973,11 +973,11 @@ func RunJoinNodeChecks(execer utilsexec.Interface, cfg *kubeadmapi.JoinConfigura
 }
 
 // RunOptionalJoinNodeChecks executes all individual, applicable to node configuration dependant checks
-func RunOptionalJoinNodeChecks(execer utilsexec.Interface, initCfg *kubeadmapi.InitConfiguration, ignorePreflightErrors sets.String) error {
+func RunOptionalJoinNodeChecks(execer utilsexec.Interface, cfg *kubeadmapi.ClusterConfiguration, ignorePreflightErrors sets.String) error {
```

Comment:
what about changing `ClusterConfiguration` to `KubeProxy` configuration only?

Reply:
We have a couple of options for this one:
1. Keep the func named `RunOptionalJoinNodeChecks` and use `ClusterConfiguration`.
2. Rename the func to `RunOptionalKubeProxyChecks` and pass `KubeProxy` configuration.

I don't think it's expected and acceptable for the func to be named `RunOptionalJoinNodeChecks` and to accept `KubeProxy` configuration directly. This is unexpected for the reader of the code and is bound to raise questions.

Reply:
we can keep this as is, as we might decide to add more checks in the function.
hopefully not, though.